src/java/org/apache/cassandra/db/virtual/GossipInfoTable.java

```diff
@@ -76,10 +89,10 @@ final class GossipInfoTable extends AbstractVirtualTable
     public DataSet data()
     {
         SimpleDataSet result = new SimpleDataSet(metadata());
-        for (Map.Entry<InetAddressAndPort, EndpointState> entry : Gossiper.instance.endpointStateMap.entrySet())
+        for (Map.Entry<InetAddressAndPort, EndpointState> entry : endpointStateMapSupplier.get().entrySet())
         {
             InetAddressAndPort endpoint = entry.getKey();
-            EndpointState localState = entry.getValue();
+            EndpointState localState = new EndpointState(entry.getValue());
```

Comment:
I think we don't need to create a new `EndpointState` here, the `EndpointState` is already copied by the `testSelectAllWithStateTransitions` test.

Reply:
@adelapena well, but imagine that this is used in production, then we get that map from the gossip. However, is not it true that by time I am populating the table, that value would change? This basically ensures that whatever happes in Gossiper, the table I am going to return basically captures one particular point in time.

Reply:
Ah, that makes sense. I thought this was related with the test fix, but it seems like a separate fix/improvement. I'm ok with it.